cloud/wfm/examples/index.ts

```diff
@@ -0,0 +1,43 @@
+import * as Promise from 'bluebird';
+import {
+  BaseTask,
+  Executor,
+  ExecutorImpl,
+  ExecutorRepository,
+  Process,
+  ProcessImpl,
+  ProcessInstance,
+  ProcessInstanceImpl,
+  Result,
+  Task,
+  TaskStatus
+} from '../src';
+
+// Derive from BaseTask to implement one that deals with custom business logic
+class MyTask extends BaseTask {
+  public run() {
+    this.status = TaskStatus.ASSIGNED;
```

Comment:
This is already a setter method in basetask: https://github.com/feedhenry-raincatcher/raincatcher-core/pull/10/files/43dd2af2593788ed5af1b99f25a933c55e9c475e#diff-07e6fd89647c2a4e0697c91d131aa024R21

it's already taking care of publishing the event, and the unit tests for Task make sure other compatible implementations should do the same: https://github.com/feedhenry-raincatcher/raincatcher-core/blob/RAINCATCH-915-wfm/cloud/wfm/test/Task/index.ts#L22